File: GDrive Integrator/facade/telegram_service.py
import requests;
import time;
from facade.env_helper import EnvironmentVariable;

def telegram_reporter(message):
    message = "["+ time.strftime("%Y-%m-%d %H:%M:%S") +"] " + message;
    
    env = EnvironmentVariable();
    bot_id = env.get_key("TELEGRAM_BOT_ID");
    chat_id = env.get_key("CHAT_ID");
    
    print(bot_id, chat_id);

    requests.request(method = "POST", 
                     url=f"https://api.telegram.org/bot{bot_id}/sendMessage?chat_id={chat_id}&text={message}", 
                     headers={}, 
                     data={}
    );

File: Pre-Processing/sr-preprocessor/Services/TelegramService.fs
namespace Services

open System;
open System.Text;
open System.Net.Http;
open Newtonsoft.Json;

open Model.Dto;
open Facade.EnvironmentVariable;

(*
    TelegramService

    This module helps the integration with Telegram. 
*)
module TelegramService =

    let SendMessage (message: string) =
    
        // Request manipulation

        let currentDateTime = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
        let requestMessage = "["+currentDateTime+"] "+message;
        let request: TelegramRequestDTO = { text = requestMessage; chat_id = EnvironmentVariable.CHAT_ID; };


        use httpRequest = new HttpClient();
        let json = JsonConvert.SerializeObject request;
        use content = new StringContent (json, Encoding.UTF8, "application/json")

        async {
            httpRequest.PostAsync("https://api.telegram.org/bot"+EnvironmentVariable.TELEGRAM_BOT_ID+"/sendMessage", content) |> Async.AwaitTask |> ignore;
        } |> Async.RunSynchronously;